### PR TITLE
Added automatic model building for logistic regression

### PR DESCRIPTION
**Pull request by ainy-123**
# Description:

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue no.) 

## Type of change:

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

 
# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

# Screenshots / Video:

### CODE REVIEW

File: flow2ml/Flow.py [hunk 1]
```diff
@@ -205,3 +206,10 @@ def getDataset(self,img_dimensions,train_val_split):
     self.update_progress( 100/100.0,"Created Datasets" ) 
 
     return self.dataset
+
+  def applyModels(self,algorithms):
+
+    self.algorithms = algorithms
+     
+    if algorithms == 'Logistic_Regression':
```
Comment (chebroluharika) [inline flow2ml/Flow.py:214]: It should be self.algorithms

File: flow2ml/Flow.py [hunk 2]
```diff
@@ -205,3 +206,10 @@ def getDataset(self,img_dimensions,train_val_split):
...
+      self.applyLogisticRegression(model)
```
Comment (chebroluharika) [inline flow2ml/Flow.py:215]: How will this work without inheriting Apply_Models class

File: flow2ml/models/model.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import pandas as pd
+import numpy as np
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV
+import os
+
+class Apply_Models:
```
Comment (chebroluharika) [inline flow2ml/models/model.py:7]: Please change this to ApplyModels(classnames should not contain underscores as per code standards)

File: flow2ml/Flow.py (hunk 1 above)
Comment (ainy-123) [reply flow2ml/Flow.py:214]: > It should be self.algorithms

There is some problem with taking model as input. I was not able to work on that part. Help me in that part plz.

File: flow2ml/models/model.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        params_grid=[{'penalty' : ['none', 'l1', 'l2', 'elasticnet'] ,
+        'C' : np.logspace(-4,4,20),
+        'solver' : ['newton-cg', 'lbfgs', 'liblinear', 'sag', 'saga'],
+        'max_iter' : [100, 1000, 2500, 5000]
+        }
+        ]
+        # Apply GridSearchCV() to find the best params.
+        grided=GridSearchCV(logmodel,param_grid=params_grid, cv=3, verbose=True, n_jobs=-1)
+        #Build model using best params on test_data
+        best_fit=grided.fit(logmodel)
```
Comment (chebroluharika) [inline flow2ml/models/model.py:26]: I didn’t understand this. 
you should try to fit model to train_data, train_labels right?
Like grided.fit(X,y) and from that you get best_params_

File: flow2ml/models/model.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        best_fit.best_estimator_
+        # return the score 
+        return best_fit.score(logmodel)
```
Comment (chebroluharika) [inline flow2ml/models/model.py:29]: After this, train model with best_params you got, then predict it with test data, then evaluate the metric that user passed as argument.

**Comment by chebroluharika** [comment]
Change filename to logistic_regression.py

File: flow2ml/Flow.py (hunk 1 above)
Comment (chebroluharika) [reply flow2ml/Flow.py:214]: @ainy-123 why do you need model as one input.

**Comment by ainy-123** [comment]
> Change filename to logistic_regression.py

we can write the function for other algorithms also under the class ApplyModels ,ryt?
Why to change the file name?
